### ISSUE TITLE
Provide wider control over who receives `form submitted` emails

### ISSUE THREAD

**Issue by signebedi** [signebedi/libreForms#94]
Right now, the system sends a verification email to the user after they submit a form. In #8, we will add more support for conditionally sending these forms / verifications to other personnel. This is a first step to that: let's create some logic to allow administrators to set their routing logic in the form config.

**Comment by signebedi** [comment]
I think we can think of three potential use cases for how we route notifications beyond just the original submitter: static list, by group, and using custom logic.

```python
        '_routing_list': {
            'type': 'static',
            'target': ['libreforms@example.com'],
        },
```
```python
        '_routing_list': {
            'type': 'groups',
            'target': ['admins'],
        },
```

```python
        '_routing_list': {
            'type': 'custom',
            'target': some_func_returning_email_list(),
        },
```

**Comment by signebedi** [comment]
Next , we add a `cc_address_list` option to `sendMail`, which expects a list of email addresses. 

What we still need is a function that can make sense of the three routing methods listed above...

**Comment by signebedi** [comment]
I'm not sure where we would place the rationalizing function, but I know that we want to route it through parse_options() as a gateway to apply defaults to missing values.


```python
def rationalize_routing_routing_list(form_name, routing_list=parse_options(form_name)['_routing_list']):
  ...

```

**Comment by signebedi** [comment]
This is pretty robustly implemented with [484b657](https://github.com/signebedi/libreForms/commit/484b657ce2da140a57bd10db8ca68424dbb463d8). There still needs to be some verification that the routing list system works under certain circumstances - for example, when there are no users in `User.users` of group `{group}`...
